pennylane/operation.py

```diff
@@ -129,6 +129,18 @@
 
 from .utils import pauli_eigs
 
+
+# =============================================================================
+# Errors
+# =============================================================================
+
+
+class NoDecompositionError(Exception):
+    """Raised when an Operator does not have a decomposition defined."""
+
+    pass
```

Comment:
Is this how you define exceptions? So simple :heart:

Reply:
We should have been doing this from the beginning!

Reply:
(we started off well with `QuantumFunctionError` and `DeviceError`, but at some point we all just started using `ValueError` 😆 )